app/views/finance/financial_transactions/new_collection.html.haml

```diff
@@ -16,16 +17,23 @@
       });
 
       $(document).on('touchclick', 'a[data-add-all-ordergroups]', function() {
-        var value = prompt("#{escape_javascript(heading_helper(FinancialTransaction, :amount))}:");
-        if (value === null)
-          return false;
+        var customField = $(this).data('custom-field');
+        var value;
+        if (!customField) {
+          value = prompt("#{escape_javascript(heading_helper(FinancialTransaction, :amount))}:");
+          if (value === null)
+            return false;
+        }
         $('#ordergroups > tbody > tr').remove();
-        var options = ordergroup.find('td > select > option').each(function() {
+        for (var id in ordergroups) {
```

Comment:
consider using `Array.forEach` (if >es5 is the target) or if backward compatible, the jquery util http://api.jquery.com/jquery.each/

otherwise, at least put `var id` at the top of the function, then `for (id in ordergroups)` - js hoisting and all that.

Reply:
and yes, i realise the rest of the js code in this project isn't super, but still, continuous improvement